core/src/main/java/com/linecorp/armeria/server/throttling/ThrottlingServiceBuilder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.throttling;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.Service;
+
+/**
+ * Builds a new {@link ThrottlingService}.
+ */
+public class ThrottlingServiceBuilder extends AbstractThrottlingServiceBuilder<HttpRequest, HttpResponse> {
+
+    ThrottlingServiceBuilder() {}
+
+    /**
+     * Sets {@link ThrottlingStrategy}.
+     */
+    public ThrottlingServiceBuilder strategy(ThrottlingStrategy<HttpRequest> strategy) {
+        setStrategy(strategy);
+        return this;
+    }
+
+    /**
+     * Sets {@link ThrottlingAcceptHandler}.
+     */
+    public ThrottlingServiceBuilder onAcceptedRequest(
+            ThrottlingAcceptHandler<HttpRequest, HttpResponse> acceptHandler) {
+        setAcceptHandler(acceptHandler);
+        return this;
+    }
+
+    /**
+     * Sets {@link ThrottlingRejectHandler}.
+     */
+    public ThrottlingServiceBuilder onRejectedRequest(
+            ThrottlingRejectHandler<HttpRequest, HttpResponse> rejectHandler) {
+        setRejectHandler(rejectHandler);
+        return this;
+    }
+
+    /**
+     * Provides default throttling reject behaviour for {@link HttpRequest}.
+     * Returns an {@link HttpResponse} with {@link HttpStatus#TOO_MANY_REQUESTS}.
+     */
+    @Override
+    protected ThrottlingRejectHandler<HttpRequest, HttpResponse> defaultRejectHandler() {
+        return (delegate, ctx, req, cause) -> HttpResponse.of(HttpStatus.TOO_MANY_REQUESTS);
```

Comment:
Let's move lambdas to singleton constants

Reply:
Moved